src/BenchmarkDotNet/ConsoleArguments/MisspellingsFinder.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.ConsoleArguments
+{
+    public class MisspellingsFinder
+    {
+        private readonly List<string> benchmarkNames = new List<string>();
+
+        //note this is an heuristic value, we suppose that user can make three misspellings
+        private static int PossibleMisspellingCount => 3;
+
+        public MisspellingsFinder(IReadOnlyList<Type> benchmarks)
+        {
+            foreach (var benchmark in benchmarks)
+            {
+                benchmarkNames.AddRange(benchmark.GetMethods()
+                    .Where(methodInfo => methodInfo.HasAttribute<BenchmarkAttribute>())
+                    .Select(methodInfo => $"{benchmark.FullName}.{methodInfo.Name}"));
+            }
+
+            benchmarkNames.AddRange(benchmarks.Select(bn => bn.FullName).Distinct());
+            benchmarkNames.AddRange(benchmarks.Select(bn => bn.Namespace).Distinct());
+        }
+
+        public string[] Find([NotNull] string userInput)
+        {
+            if (userInput == null) 
+                throw new ArgumentNullException(nameof(userInput));
+            return benchmarkNames.Where(name => GetLevenshteinDistance(userInput, name) <= PossibleMisspellingCount).ToArray();
+        }
+
+        private static int GetLevenshteinDistance(string string1, string string2)
```

Comment:
Could you write a few unit tests for `GetLevenshteinDistance`? (just for pure string input without the reflection logic).